src/reps/array.js

```diff
@@ -19,184 +19,114 @@ const DOM = React.DOM;
  * Renders an array. The array is enclosed by left and right bracket
  * and the max number of rendered items depends on the current mode.
  */
-let ArrayRep = React.createClass({
-  displayName: "ArrayRep",
```

Comment:
Ah great to know where this is coming from at least. Totally agree with your proposal, let's check if there is a performance impact. If there is one, yes flow will be an option.